### PR TITLE
Azure-Core pyright clean and enabled

### PR DESCRIPTION
**Pull request by lmazuel**

### CODE REVIEW

File: sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared.py [hunk 1]
```diff
@@ -287,16 +336,38 @@ def _get_raw_parts_helper(response, http_response_type: Type):
...
+    response: PipelineTransportHttpResponse,
+) -> List[PipelineTransportHttpResponse]:
+    ...
+
+
 def _parts_helper(
-    response: "PipelineTransportHttpResponse",
-) -> Iterator["PipelineTransportHttpResponse"]:
+    response: Any,
+) -> List[Any]:
```
Comment (kashifkhan) [inline sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared.py:370]: Will some of the typing for `_get_raw_parts` ( and chain of calls ) have to be updated from `Iterator` to `List` as well ?

Comment (kashifkhan) [reply sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared.py:370]: nvm maybe just `_get_raw_parts` as helper has been updated above

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 2]
```diff
@@ -519,7 +526,7 @@ def stream_download(self, pipeline: Pipeline[HttpRequest, "HttpResponse"], **kwa
         """
         raise NotImplementedError("stream_download is not implemented.")
 
-    def parts(self) -> Iterator["HttpResponse"]:
+    def parts(self) -> Iterable["HttpResponse"]:
```
Comment (kashifkhan) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:529]: is this `Iterable` if other overloads of _parts_helper get added later on returning other types?

File: sdk/core/azure-core/azure/core/pipeline/transport/_base_async.py [hunk 3]
```diff
@@ -94,7 +94,7 @@ def stream_download(
         """
         raise NotImplementedError("stream_download is not implemented.")
 
-    def parts(self) -> AsyncIterator["AsyncHttpResponse"]:
+    def parts(self) -> AsyncIterable["AsyncHttpResponse"]:
         """Assuming the content-type is multipart/mixed, will return the parts as an async iterator.
```
Comment (kashifkhan) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base_async.py:98]:
```suggestion
        """Assuming the content-type is multipart/mixed, will return the parts as an async iterable.
```

**Review by kristapratico — APPROVED** [review]
🚀

**Pull request closed without merging**